cloudify_types/cloudify_types/component/operations.py

```diff
@@ -282,6 +301,17 @@ def create(timeout=EXECUTIONS_TIMEOUT, interval=POLLING_INTERVAL, **kwargs):
     # TODO capabilities are unused?
     # deployment_capabilities = deployment.get('capabilities')
     deployment_auto_suffix = deployment.get('auto_inc_suffix', False)
+    deployment_labels = deployment.get('labels', [])
+    if not validate_labels(deployment_labels):
+        raise NonRecoverableError(
+            "The provided deployment labels are not valid. "
+            "Labels must be a list of single-entry dicts, "
+            "e.g. [{\'foo\': \'bar\'}]. "
+            "This value was provided: %s." % deployment_labels
+        )
+
+    if not any('csys-obj-parent' in item for item in deployment_labels):
+        deployment_labels.append({'csys-obj-parent': ctx.deployment.id})
```

Comment:
Is it safe to assume that all components should get `csys-obj-parent` label set by default to `ctx.deployment.id` (unless specified otherwise)?